From c02c7bbadf6d97f3d178289909495483b66a921f Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Fri, 14 Nov 2025 12:42:46 +0100
Subject: [PATCH 1/6] arch: arm: mach-socfpga: Fix non-Agilex targets

The Altera system manager device is only provided for the Agilex
targets. All others were left with an uninitialized socfpga_sysmgr_base
after that refactoring, breaking any boot attempts already in SPL.

Tested only on Cyclone5 / DE0-Nano-SoC.

Fixes: 35638172f99a ("arm: socfpga: agilex5: Add new driver model for system manager in Agilex5")
Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 arch/arm/mach-socfpga/misc.c     | 7 +++++++
 arch/arm/mach-socfpga/spl_a10.c  | 1 +
 arch/arm/mach-socfpga/spl_gen5.c | 1 +
 arch/arm/mach-socfpga/spl_n5x.c  | 1 +
 arch/arm/mach-socfpga/spl_s10.c  | 1 +
 5 files changed, 11 insertions(+)

diff --git a/arch/arm/mach-socfpga/misc.c b/arch/arm/mach-socfpga/misc.c
index 97e01140513..dee822fa870 100644
--- a/arch/arm/mach-socfpga/misc.c
+++ b/arch/arm/mach-socfpga/misc.c
@@ -277,6 +277,8 @@ void socfpga_get_managers_addr(void)
 void socfpga_get_sys_mgr_addr(const char *compat)
 {
 	int ret;
+#if defined(CONFIG_TARGET_SOCFPGA_AGILEX) || \
+    defined(TARGET_SOCFPGA_AGILEX7M) || defined(TARGET_SOCFPGA_AGILEX5)
 	struct udevice *sysmgr_dev;
 
 	ret = uclass_get_device_by_name(UCLASS_NOP, compat, &sysmgr_dev);
@@ -286,6 +288,11 @@ void socfpga_get_sys_mgr_addr(const char *compat)
 	} else {
 		socfpga_sysmgr_base = (phys_addr_t)dev_read_addr(sysmgr_dev);
 	}
+#else
+	ret = socfpga_get_base_addr("altr,sys-mgr", &socfpga_sysmgr_base);
+	if (ret)
+		hang();
+#endif
 }
 
 phys_addr_t socfpga_get_rstmgr_addr(void)
diff --git a/arch/arm/mach-socfpga/spl_a10.c b/arch/arm/mach-socfpga/spl_a10.c
index c20376f7f8e..6722c9cef3f 100644
--- a/arch/arm/mach-socfpga/spl_a10.c
+++ b/arch/arm/mach-socfpga/spl_a10.c
@@ -250,6 +250,7 @@ void board_init_f(ulong dummy)
 	if (spl_early_init())
 		hang();
 
+	socfpga_get_sys_mgr_addr(NULL);
 	socfpga_get_managers_addr();
 
 	dcache_disable();
diff --git a/arch/arm/mach-socfpga/spl_gen5.c b/arch/arm/mach-socfpga/spl_gen5.c
index df79cfe0f7f..cd8b09d7a8d 100644
--- a/arch/arm/mach-socfpga/spl_gen5.c
+++ b/arch/arm/mach-socfpga/spl_gen5.c
@@ -72,6 +72,7 @@ void board_init_f(ulong dummy)
 	if (ret)
 		hang();
 
+	socfpga_get_sys_mgr_addr(NULL);
 	socfpga_get_managers_addr();
 
 	/*
diff --git a/arch/arm/mach-socfpga/spl_n5x.c b/arch/arm/mach-socfpga/spl_n5x.c
index 81283ef7162..2b6a443e989 100644
--- a/arch/arm/mach-socfpga/spl_n5x.c
+++ b/arch/arm/mach-socfpga/spl_n5x.c
@@ -31,6 +31,7 @@ void board_init_f(ulong dummy)
 	if (ret)
 		hang();
 
+	socfpga_get_sys_mgr_addr(NULL);
 	socfpga_get_managers_addr();
 
 	/* Ensure watchdog is paused when debugging is happening */
diff --git a/arch/arm/mach-socfpga/spl_s10.c b/arch/arm/mach-socfpga/spl_s10.c
index fa83ff96adc..379e5642b2a 100644
--- a/arch/arm/mach-socfpga/spl_s10.c
+++ b/arch/arm/mach-socfpga/spl_s10.c
@@ -33,6 +33,7 @@ void board_init_f(ulong dummy)
 	if (ret)
 		hang();
 
+	socfpga_get_sys_mgr_addr(NULL);
 	socfpga_get_managers_addr();
 
 	/* Ensure watchdog is paused when debugging is happening */
-- 
2.51.0

